source/replication/site-replication-overview.rst

```diff
@@ -0,0 +1,462 @@
+.. _minio-site-replication-overview:
+
+=========================
+Site replication overview
+=========================
+
+.. default-domain:: minio
+
+.. contents:: Table of Contents
+   :local:
+   :depth: 1
+
+Site replication configures multiple independent MinIO sites or clusters as replicas.
```

Comment:
I've been trying to use `MinIO Deployments` as the preferred term over 'clusters' or 'sites', though we should check w/ engineering and make sure we are completely consistent in usage here.